addons/stock/models/stock_move.py

```diff
@@ -1261,8 +1261,10 @@ def _action_done(self, cancel_backorder=False):
         if self.env.context.get('is_scrap'):
             return moves_todo
 
-        if picking and not cancel_backorder:
-            picking._create_backorder()
+        if not cancel_backorder:
+            pickings_to_backorder = pickings._get_pickings_to_backorder()
```

Comment:
It's probably useless to check for pickings to backorder since the wizard do not allow to filter on which pickings should have a backorder or not. pickings._create_backorder() should be sufficient